pkg/registry/apis/query/query.go

```diff
@@ -153,12 +153,21 @@ func (b *QueryAPIBuilder) execute(ctx context.Context, req parsedRequestInfo) (q
 		break // nothing to do
 	case 1:
 		qdr, err = b.handleQuerySingleDatasource(ctx, req.Requests[0])
+		if err != nil {
+			return
+		}
 	default:
 		qdr, err = b.executeConcurrentQueries(ctx, req.Requests)
+		if err != nil {
```

Comment:
it feels like the current behavior might be intentional to support partial results. maybe that is only relevant for queries that originate from the UI? @ryantxu might have more context here

Reply:
I wonder how this would work, as you would always get this error and not data? I might miss some context around this though :-D

Reply:
here's an example response for running two queries where one fails. the HTTP response code is 400 because of the query parsing error, but a partial response is returned:
```json
{
  "results": {
    "A": {
      "status": 200,
      "frames": [
        {
          "schema": {
            "refId": "A",
            "meta": {
              "type": "timeseries-multi",
              "typeVersion": [
                0,
                1
              ],
              "custom": {
                "resultType": "matrix"
              },
              "executedQueryString": "Expr: vector(1)\nStep: 30s"
            },
            "fields": [
              {
                "name": "Time",
                "type": "time",
                "typeInfo": {
                  "frame": "time.Time"
                },
                "config": {
                  "interval": 30000
                }
              },
              {
                "name": "Value",
                "type": "number",
                "typeInfo": {
                  "frame": "float64"
                },
                "labels": {},
                "config": {
                  "displayNameFromDS": "vector(1)"
                }
              }
            ]
          },
          "data": {
            "values": [
              [
                1720187070000,
                1720187100000,
                1720187130000,
                1720187160000,
                1720187190000,
                1720187220000,
                1720187250000,
                1720187280000,
                1720187310000,
                1720187340000,
                1720187370000,
                1720187400000,
                1720187430000,
                1720187460000,
                1720187490000,
                1720187520000,
                1720187550000,
                1720187580000,
                1720187610000,
                1720187640000,
                1720187670000,
                1720187700000
              ],
              [
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1,
                1
              ]
            ]
          }
        }
      ]
    },
    "B": {
      "error": "bad_data: invalid parameter \"query\": 1:1: parse error: unknown function with name \"foo\"",
      "errorSource": "",
      "status": 400,
      "frames": [
        {
          "schema": {
            "refId": "B",
            "meta": {
              "typeVersion": [
                0,
                0
              ],
              "executedQueryString": "Expr: foo(1)\nStep: 30s"
            },
            "fields": []
          },
          "data": {
            "values": []
          }
        }
      ]
    }
  }
}
```

Reply:
Ah yeah I see, that would be i.e. the explore view in Grafana. Got it. I think we could check that we only return early if we got expressions as part of the request.

Reply:
here's the current behavior with an expression in the mix:
<img width="549" alt="image" src="https://github.com/grafana/grafana/assets/360020/4040b027-e9dc-4aa6-b846-1a1eb6ca6807">

is the main issue that we are returning an HTTP error instead of the partial results in the new query service?

Reply:
maybe we could avoid returning the error here if the response is set? https://github.com/grafana/grafana/blob/5fcbe871163501af11eeb779b048738ee3c1444b/pkg/registry/apis/query/query.go#L137-L142